WordPress/src/main/java/org/wordpress/android/ui/mysite/personalization/PersonalizationViewModel.kt

```diff
@@ -22,153 +14,31 @@ const val CARD_TYPE_TRACK_PARAM = "type"
-            }
-            // update the ui state
-            updateCardState(cardType, enabled)
-        }
-    }
-
-    private fun trackCardToggle(cardType: CardType, enabled: Boolean){
-        if(enabled) trackCardShown(cardType)
-        else trackCardHidden(cardType)
-    }
-
-    private fun trackCardHidden(cardType: CardType){
-        analyticsTrackerWrapper.track(
-            Stat.PERSONALIZATION_SCREEN_CARD_HIDE_TAPPED,
-            mapOf(CARD_TYPE_TRACK_PARAM to cardType.trackingName)
-        )
+        dashboardCardPersonalizationViewModelSlice.start(siteId)
+        shortcutsPersonalizationViewModelSlice.start(selectedSiteRepository.getSelectedSite()!!)
     }
 
-    private fun trackCardShown(cardType: CardType){
-        analyticsTrackerWrapper.track(
-            Stat.PERSONALIZATION_SCREEN_CARD_SHOW_TAPPED,
-            mapOf(CARD_TYPE_TRACK_PARAM to cardType.trackingName)
-        )
-    }
-
-    private fun updateCardState(cardType: CardType, enabled: Boolean) {
-        val currentCards: MutableList<DashboardCardState> = _uiState.value!!.toMutableList()
-        val updated = currentCards.find { it.cardType == cardType }!!.copy(enabled = enabled)
-        currentCards[cardType.order] = updated
-        _uiState.postValue(currentCards)
+    fun onCardToggled(cardType: CardType, enabled: Boolean) {
+        dashboardCardPersonalizationViewModelSlice.onCardToggled(cardType, enabled)
     }
 
-    private fun isStatsCardShown(siteId: Long) = !appPrefsWrapper.getShouldHideTodaysStatsDashboardCard(siteId)
-
-    private fun isDraftPostsCardShown(siteId: Long) =
-        !appPrefsWrapper.getShouldHidePostDashboardCard(siteId, PostCardType.DRAFT.name)
-
-    private fun isScheduledPostsCardShown(siteId: Long) =
-        !appPrefsWrapper.getShouldHidePostDashboardCard(siteId, PostCardType.SCHEDULED.name)
-
-    private fun isPagesCardShown(siteId: Long) = !appPrefsWrapper.getShouldHidePagesDashboardCard(siteId)
-
-    private fun isActivityLogCardShown(siteId: Long) = !appPrefsWrapper.getShouldHideActivityDashboardCard(siteId)
-
-    private fun isBlazeCardShown(siteId: Long) = !appPrefsWrapper.hideBlazeCard(siteId)
-
-    private fun isNextStepCardShown(siteId: Long) = !appPrefsWrapper.getShouldHideNextStepsDashboardCard(siteId)
-
-    private suspend fun isPromptsSettingEnabled(
-        siteId: Int
-    ): Boolean = bloggingRemindersStore
-        .bloggingRemindersModel(siteId)
-        .firstOrNull()
-        ?.isPromptsCardEnabled == true
-
-    private suspend fun updatePromptsCardEnabled(isEnabled: Boolean) {
-        val siteId = selectedSiteRepository.getSelectedSiteLocalId()
-        val current = bloggingRemindersStore.bloggingRemindersModel(siteId).firstOrNull() ?: return
-        bloggingRemindersStore.updateBloggingReminders(current.copy(isPromptsCardEnabled = isEnabled))
+    override fun onCleared() {
+        super.onCleared()
+        shortcutsPersonalizationViewModelSlice.onCleared()
```

Comment:
❤️  Thanks for adding this. Now we can be sure that the scopes are canceled in the viewModel slices.